go/adbc/driver/databricks/schema.go

```diff
@@ -29,6 +29,11 @@ import (
 
 const DbxSchemaTypeText = "type_text"
 
+const (
+	decimalTypeRegex  = `^(?:DECIMAL|DEC|NUMERIC)\((\d+)(?:,(\d+))?\)$`
+	intervalTypeRegex = `^INTERVAL\s+(YEAR(?:\s+TO\s+MONTH)?|MONTH|DAY(?:\s+TO\s+(HOUR|MINUTE|SECOND))?|HOUR(?:\s+TO\s+(MINUTE|SECOND))?|MINUTE(?:\s+TO\s+SECOND)?|SECOND)`
```

Comment:
This is specifically the SQL engine which does normalize upward, but one edge case here is to use case insensitive matching for protective coverage. May _not_ be needed though. It's probably worth just double checking if any code paths in this driver reach for a thrift env or similar, as those would require upcasing.